examples/3.routes/8.tolls-and-ferries/client.js

```diff
@@ -48,12 +48,12 @@ export const getRoute = callback => {
         client.executeSubscription(createRequest(routeUpdateSubscription, { id: routeId })),
         subscribe(result => {
           const { status, route } = result.data?.routeUpdatedById;
-          // You can keep listening to the route changes to update route information.
-          // For this example we want to only draw the initial route.
+          // Subscription used as computation time can increase for longer routes.
           if (status === 'done' && route) {
             unsubscribe();
             callback(route);
-          } else if (status === 'not_found') {
+          } else if (status === 'not_found' || status === 'error') {
+            unsubscribe();
             callback();
```

Comment:
We never do anything with an empty callback, think we can just unsubscribe right?

Reply:
And we might want to move unsubscribe to when the status is not pending. But not sure if that's any better than what we currently do. Leave it up to you.